### PR TITLE
Fix wrong stepkey for nextKey in StepTestCase#mutateInstance(...)

### PR DESCRIPTION
**Pull request by mushao999**
This pr fix incorrect stepKey for nextKey in StepTestCase#mutateInstance(...).

### CODE REVIEW

File: docs/changelog/89494.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pr: 89494
```
Comment (gmarouli) [inline docs/changelog/89494.yaml:1]: I do not think we need a change log for this since this is a test change. Do you mind removing it?

**Review by gmarouli — CHANGES_REQUESTED** [review]
@mushao999 thank you for your contribution and your thoroughness catching all the instances of this bug. Just a request to remove the change log and then I will merge!

**Comment by gmarouli** [comment]
@elasticmachine ok to test

File: docs/changelog/89494.yaml (hunk 1 above)
Comment (mushao999) [reply docs/changelog/89494.yaml]: Thanks for your reply, I've removed the changelog

Comment (gmarouli) [reply docs/changelog/89494.yaml]: Great, thanks!

**Review by gmarouli — APPROVED** [review]
@mushao999 , LGTM, once again thank you for your contribution and the quick reaction to the comments! I will move forward with merging this.

**Pull request merged**